# Week 1 Genome Assembly -- Feedback

1 + 1 + 1 + 1 + 1 + 1 + 0.75 + 0.5775 + 0.5 + 1  = 8.83 points out of 10 possible points

1. Question 1.1, 1.4 how many reads (0.5 pts each)

  * good work --> +1

2. Question 1.2, 1.4 simulation script(s)

  * nice script --> +1; just a note that you can use the size argument to specify the number of reads you want to simulate and therefore simulate more than one read at a time

3. Question 1.2, 1.4 plotting script(s)

  * --> + 1

4. Question 1.2, 1.4 histograms with overlaid Poisson distributions (0.5 pts each)

  * good work overall; if you would like a height matchup with the poisson distribution and the histogram, you could change the number of bins used in the histogram to match the number of unique coverage whole number values plotted/represented in the dataset --> + 1

5. Question 1.3, 1.4 how much of genome not sequenced/comparison to Poisson expectations (0.5 pts each, 0.25 for answer and 0.25 for code)

  * good work overall and your for loop is a nice way to find the answer for the observed value, but where is the code/how did you find out the Poisson expectation?  --> +0.5

6. Question 2 De novo assembly (0.5 pts each, 0.25 for answer and 0.25 for code)

  * number of contigs
    * --> +0.5
  * total length of contigs
    * very nice solution --> +0.5

7. Question 2 De novo assembly cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * size of largest contig
    * --> + 0.5
  * contig n50 size
    * how did you find this? --> + 0.25

8. whole genome alignment (0.33 pts each, 0.33/2 for answer and 0.33/2 for code)

  * average identity
    * --> + 0.33
  * length of longest alignment
    * I think the longest alignment is a different number; please show or explain where you're getting that number from the out report --> + 0
  * how many insertions and deletions in assembly
    * how many deletions were there? --> 0.2475 (+ 0.33/4 for half answer + 0.33/2 for explanation/code)

9. decoding the insertion (0.5 pts each, 0.25 for answer and 0.25 for code)

  * position of insertion in assembly
    * --> + 0.5
  * length of novel insertion
    * close, but not quite; how did you find the length? --> + 0

10. decoding the insertion cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * DNA sequence of encoded message
    * --> +0.5
  * secret message
    * --> +0.5
